File: spec/dummy/app/matestack/components/my_form_input.rb
class Components::MyFormInput < Matestack::Ui::Core::Form::Input::Base

  vue_js_component_name "my-form-input"

  def prepare
    # optionally add some data here, which will be accessible within your Vue.js component
    @component_config[:foo] = "bar"
  end

  def response
    div do
      label text: "my form input"
      input input_attributes.merge(class: "flatpickr")
      render_errors
    end
  end

end

File: spec/dummy/app/matestack/components/my_form_input.js
MatestackUiCore.Vue.component('my-form-input', {
  mixins: [MatestackUiCore.componentMixin, MatestackUiCore.formInputMixin],
  data() {
    return {
    };
  },
  methods: {

  },
  mounted: function(){
    flatpickr(this.$el.querySelector('.flatpickr'), {
      defaultDate: this.componentConfig["init_value"],
      enableTime: true
    });
  }
});
